coderd/entitlements/entitlements.go

```diff
@@ -0,0 +1,110 @@
+package entitlements
+
+import (
+	"encoding/json"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/coder/coder/v2/codersdk"
+)
+
+type Set struct {
```

Comment:
I implemented the methods as I saw them used. There might be a way to reduce the number of methods on this struct.